.clang-tidy

```diff
@@ -12,12 +12,17 @@ Checks: >-
   performance-*,
   portability-*,
   readability-*,
+  -clang-analyzer-core.uninitialized.Assign,
   -cppcoreguidelines-avoid-const-or-ref-data-members,
+  -google-readability-avoid-underscore-in-googletest-name,
   -google-readability-todo,
+  -llvm-header-guard,
+  -modernize-concat-nested-namespaces,
+  -modernize-type-traits,
+  -modernize-use-constraints,
+  -modernize-use-nodiscard,
```

Comment:
Shouldn't we be using our `CETL_NODISCARD` where it says nodiscard is needed?

Reply:
`CETL_NODISCARD` is not working for c++14 - this rule expects specifically `[[nodiscard]]`